### PR TITLE
tests: benchmarks: Verify power management for TWIM (using s2ram)

### PR DESCRIPTION
**Pull request by nordic-bami**
Verify PM ability to handle TWIM power management
This sample uses pca63566 shield with bme688 i2c sensor

### CODE REVIEW

File: tests/benchmarks/multicore/idle_twim/src/main.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{
+	uint8_t response;
+	uint32_t i2c_config = I2C_SPEED_SET(I2C_SPEED_STANDARD) | I2C_MODE_CONTROLLER;
+
+	printk("Device address 0x%x\n", DEVICE_ADDRESS);
+	printk("I2C speed setting: %d\n", I2C_SPEED_STANDARD);
+
+	response = i2c_configure(i2c_device, i2c_config);
+	if (response > 0) {
+		printk("I2C configuration failed%d\n", response);
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_twim/src/main.c:47]: return on error?

File: tests/benchmarks/multicore/idle_twim/src/main.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	while (1) {
+		response = read_sensor_register(CHIP_ID_REGISTER_ADDRESS);
+		printk("Chip_Id: %d\n", response);
+
+		response = read_sensor_register(VARIANT_ID_REGISTER_ADDRESS);
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_twim/src/main.c:54]: can response be verified and return at error?

Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_twim/src/main.c:54]: I would not do that, chip ID may change in the future (if for example there will be new sensor boards).

File: tests/benchmarks/multicore/idle_twim/src/main.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+int main(void)
+{
+	uint8_t response;
+	uint32_t i2c_config = I2C_SPEED_SET(I2C_SPEED_STANDARD) | I2C_MODE_CONTROLLER;
+
+	printk("Device address 0x%x\n", DEVICE_ADDRESS);
+	printk("I2C speed setting: %d\n", I2C_SPEED_STANDARD);
+
+	response = i2c_configure(i2c_device, i2c_config);
+	if (response > 0) {
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_twim/src/main.c:46]: error is response < 0?

Comment (nordic-piks) [reply tests/benchmarks/multicore/idle_twim/src/main.c:46]: and return type int?

File: tests/benchmarks/multicore/idle_twim/src/main.c (hunk 2 above)
Comment (nordic-piks) [reply tests/benchmarks/multicore/idle_twim/src/main.c:55]: Maybe just checking if it > 0?

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-pikr — APPROVED** [review]

File: tests/benchmarks/multicore/idle_twim/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+if (NOT SYSBUILD)
+	message(WARNING
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_twim/CMakeLists.txt:11]: FATAL_ERROR if sysbuild is required

File: tests/benchmarks/multicore/idle_twim/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+if (NOT SYSBUILD)
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_twim/CMakeLists.txt:10]:
```suggestion
if(NOT SYSBUILD)
```
No spaces between these in cmake

File: tests/benchmarks/multicore/idle_twim/boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        s2ram: s2ram {
+            compatible = "zephyr,power-state";
+            power-state-name = "suspend-to-ram";
+            min-residency-us = <800000>;
+        };
+    };
+};
+
+&cpu {
+    cpu-power-states = <&idle &s2ram>;
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_twim/boards/nrf54h20dk_nrf54h20_cpuapp_s2ram.overlay:24]: tab indent for dts

File: tests/benchmarks/multicore/idle_twim/testcase.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+common:
+  sysbuild: true
+  depends_on: i2c
+  tags: ci_build ci_tests_benchmarks_multicore twim
+
+
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_twim/testcase.yaml:6]:
```suggestion
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: tests/benchmarks/multicore/idle_twim/CMakeLists.txt (hunk 4 above)
Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_twim/CMakeLists.txt:11]: Why, existing samples in benchmarks have all the same configuration with 'warning' ?

Comment (nordicjm) [reply tests/benchmarks/multicore/idle_twim/CMakeLists.txt:11]: If it needs sysbuild to work, then fatal error, if it does not, then warning. Also don't mark conversations as resolved without either implementing the fix or the other person dismissing their own comments, thanks.

**Review by nordicjm — COMMENTED** [review]
you've made changes to a different test, but not the test in question in this PR?

**Comment by nordic-bami** [comment]
> you've made changes to a different test, but not the test in question in this PR?
Reverted the incorrect changed and applied in right place now

**Review by nordicjm — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Pull request merged**